nats.go

```diff
@@ -1995,26 +1996,17 @@ func (nc *Conn) respHandler(m *Msg) {
 // Create the response subscription we will use for all
 // new style responses. This will be on an _INBOX with an
 // additional terminal token. The subscription will be on
-// a wildcard.
+// a wildcard. Caller is responsible for synchronization.
 func (nc *Conn) createRespMux() error {
 	// _INBOX wildcard
 	ginbox := fmt.Sprintf("%s.*", NewInbox())
 	s, err := nc.Subscribe(ginbox, nc.respHandler)
 	if err != nil {
 		return err
 	}
-	// We could be racing here. So will we double check
-	// respMux here and discard the new one if set.
-	nc.mu.Lock()
-	if nc.respMux == nil {
-		nc.respSub = ginbox
-		nc.respMux = s
-		nc.respMap = make(map[string]chan *Msg)
-	} else {
-		// Discard duplicate, don't set others.
-		defer s.Unsubscribe()
-	}
-	nc.mu.Unlock()
+	nc.respSub = ginbox
```

Comment:
Should we have locking around this? What if the connection is closed from another go-routine?